core/backend/src/IModelTransformer.ts

```diff
@@ -939,11 +944,20 @@ export class TemplateModelCloner extends IModelTransformer {
     const predecessorIds: Id64Set = sourceElement.getPredecessorIds();
     predecessorIds.forEach((predecessorId: Id64String) => {
       if (Id64.invalid === this.context.findTargetElementId(predecessorId)) {
-        throw new IModelError(IModelStatus.BadRequest, "Required dependency not found in target iModel", Logger.logError, BackendLoggerCategory.IModelTransformer);
+        if (this.context.isBetweenIModels) {
+          throw new IModelError(IModelStatus.BadRequest, "Required dependency not found in target iModel", Logger.logError, BackendLoggerCategory.IModelTransformer);
+        } else {
+          const definitionElement = this.sourceDb.elements.tryGetElement<DefinitionElement>(predecessorId, DefinitionElement);
+          if (definitionElement && !(definitionElement instanceof RecipeDefinitionElement)) {
+            this.context.remapElement(predecessorId, predecessorId); // when in the same iModel, can use existing DefinitionElements without remapping
+          } else {
+            throw new IModelError(IModelStatus.BadRequest, "Required dependency not found", Logger.logError, BackendLoggerCategory.IModelTransformer);

```

Comment:
Include `predecessorId` in exception message?

Reply:
Done.